### PR TITLE
Update to Ruby 3

### PR DESCRIPTION
**Pull request by kirillplatonov**
- Updates default Ruby version as 2.5 is not maintained anymore
- Adds Ruby 3 to CI

### CODE REVIEW

**Comment by kirillplatonov** [comment]
cc @NabeelAhsen

**Comment by kirillplatonov** [comment]
@gonzaloriestra updated. Ready for merge.

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -12,7 +12,7 @@ jobs:
     name: Ruby ${{ matrix.version }}
     strategy:
       matrix:
-        version: [2.5, 2.6, 2.7]
+        version: [2.5, 2.6, 2.7, '3.0']
```
Comment (gonzaloriestra) [inline .github/workflows/build.yml:15]: Let's better remove 2.5 to just keep supported versions. Also, does 3.0 require the quotes? 🤔 

```suggestion
        version: [2.6, 2.7, 3.0]
```

**Review by gonzaloriestra — APPROVED** [review]
Thanks for your help! ❤️

File: .github/workflows/build.yml (hunk 1 above)
Comment (kirillplatonov) [reply .github/workflows/build.yml]: Removed 2.5. Quotes for `3.0` are required due to a bug in Github Actions runner: https://github.com/actions/runner/issues/849

**Comment by gonzaloriestra** [comment]
I'm trying to merge this, but I cannot remove the Ruby 2.5 check. I've tried to create another PR, but it's still there... I guess we need to update the project settings, but I don't have permission, so I'll ask for help.

If you don't mind, we can close here and continue in https://github.com/Shopify/shopify_app/pull/1359, which includes a couple of more things (I've added you as a co-author).

**Pull request closed without merging**